### DISCUSSION TITLE
How to add Steam Guard to Authenticator?

### DISCUSSION THREAD

**Discussion by TBloop** [Authenticator-Extension/Authenticator#891] [Q&A]
I noticed that the description for Authenticator says it supports steam guard, however I'm unsure how to add it to the extension. Steam doesn't give you a QR code or anything like that when you enable it on your phone. 

I know it's possible to add codes manually using certain parameters but I'm unsure how to get them. 

Could you please point me in the right direction on how to get Steam Guard added?

Thank you!

**Comment by TBloop** [comment]
I figured it out. I don't know if I should delete this thread but I will share the solution I found.

My initial goal was to have Steam Guard on both my phone and also the extension, however with this method I couldn't also have it on my phone. Having it on your phone requires getting the "secret" key which apparently can be done but is confusing.

Follow the [install & setup instructions for the Steam Desktop Authenticator](https://github.com/Jessecar96/SteamDesktopAuthenticator). This program can generate the codes by itself, but I wanted them in the extension. Do not encrypt/add a password during setup.

Look in the maFiles subdirectory where the SDA is installed. One of the files will be named [your_steamID].maFile. Open it.

One of the JSON variables will look like: "uri":"otpauth://totp/Steam:your-username?secret=ABCDEFGHIJKLMN1234OPQRSTUVWXYZ4321&issuer=Steam"

You are looking for the part after secret= like this: ABCDEFGHIJKLMN1234OPQRSTUVWXYZ4321

Now, go into the extension, press the pencil edit button on top right, then the add button, then press manual entry.

1. Issuer: Steam
2. Secret: Your secret from before goes here
3. Expand the "advanced" arrow
4. Username: Your steam username
5. Period: leave this blank
6. Digits: 6 (I left this default)
7. Algorithm: SHA-1 (I left this default)
8. Type: steam

Then press ok. For me I think it displayed numbers at first, but after I re-opened it a minute later it was displaying the letters and numbers properly and it matched Steam Desktop Authenticator. 

The problem is, you cant have SDA enabled and also the Steam app on your phone. So I disabled SDA and re-enabled Steam Authenticator on my phone, and so I couldn't retrieve the secret key, and therefore can't add it to the extension. There are apparently ways to get it but I'll leave that to you to google.

**Reply by sosawolfo** [reply]
@TBloop thank you so much! that worked!

**Reply by thomas-rhong** [reply]
thank you. i found it numbers so i thought it wrong and didn't open again.

**Reply by TrevorLaneRay** [reply]
I normally avoid necro-posting, but can confirm this **does still work**.
Verified it with two YubiKeys (firmware versions 5.7.1 and 5.4.3) and with BitWarden as well.
Kudos.
